### PR TITLE
Update DeDup Module

### PR DESCRIPTION
**Pull request by jfy133**
This updates DeDup to match latest [0.12.6](https://github.com/apeltzer/DeDup/releases/tag/0.12.6) release.

Changes include reporting in general stats table of mapped reads after DeDuplication.
Bar plot represents total mapped reads, rather than all reads in BAM file, as DeDup only works with mapped reads.

## If this PR is _not_ a new module
 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` has been updated
 - [x] (optional but recommended): https://github.com/ewels/MultiQC_TestData contains test data for this change

## If this PR is for a new module
 - [ ] There is example tool output for tools in the https://github.com/ewels/MultiQC_TestData repository
 - [ ] Code is tested and works locally (including with `--lint` flag)
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated
 - [ ] `docs/README.md` is updated with link to below
 - [ ] `docs/modulename.md` is created
 - [ ] Everything that can be represented with a plot instead of a table is a plot
 - [ ] Report sections have a description and help text (with `self.add_section`)
 - [ ] There aren't any huge tables with > 6 columns (explain reasoning if so)
 - [ ] Each table column has a different colour scale to its neighbour, which relates to the data (eg. if high numbers are bad, they're red)
 - [ ] Module does not do any significant computational work

### CODE REVIEW

**Comment by jfy133** [comment]
Test data: https://github.com/jfy133/MultiQC_TestData/blob/master/data/modules/dedup/v0.12.6/JK2782.mapped.dedup.json

@apeltzer please check new calculations/plotting is right!

**Comment by jfy133** [comment]
Test data: https://github.com/ewels/MultiQC_TestData/pull/150

**Comment by apeltzer** [comment]
@ewels Just so you know: As we're probably the only ones using DeDup + MultiQC in combination right now and EAGERv2.2.0 will be the first "publishable" version of the eager pipeline, we don't need to maintain downward compatibility with DeDup v0.12.5 and earlier.

**Comment by ewels** [comment]
hah, ok - if you say so 😉

**Comment by jfy133** [comment]
@ewels I wonder if tests failed because you merged the newest test data after I made this PR (because of my left over empty file). Could you re-run tests?

**Comment by jfy133** [comment]
@ewels ...help!?!?!?!!?

image

**Comment by ewels** [comment]
Manually restarted GHA, that should fix it. A lot of PRs have started randomly timing out. I wonder if there is some new test data that is taking ages to process or something 😕

**Comment by jfy133** [comment]
@ewels same error message

```
[INFO   ]         biscuit : Found 66 reports
[CRITICAL]         multiqc : User Cancelled Execution!
============================================================
Traceback (most recent call last):
  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\multiqc\multiqc.py", line 556, in run
    output = mod()
  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\multiqc\modules\hicexplorer\hicexplorer.py", line 32, in __init__
    self.write_data_file(self.mod_data, 'multiqc_hicexplorer')
  File "c:\hostedtoolcache\windows\python\3.6.8\x64\lib\site-packages\multiqc\modules\base_module.py", line 338, in write_data_file
    fn = '{}_{}'.format(base_fn, i)
KeyboardInterrupt
============================================================
User Cancelled Execution!
Exiting MultiQC...
##[error]The operation was canceled.
```

It's the same errors for #1179

**Comment by ewels** [comment]
Ok, I think I found the problem. I did some refactoring in HiCExporer fairly recently and there was a bit of dodgy code there. Hopefully fixed now 🤞 

Phil

**Comment by ewels** [comment]
ok, phew - got the tests passing again ✅

File: multiqc/modules/dedup/dedup.py [hunk 1]
```diff
@@ -107,14 +121,29 @@ def dedup_general_stats_table(self):
...
+        }
         self.general_stats_addcols(self.dedup_data, headers)
 
     def dedup_alignment_plot (self):
         """ Make the HighCharts HTML to plot the duplication rates """
 
         # Specify the order of the different possible categories
         keys = OrderedDict()
-        keys['not_removed'] = { 'name': 'Not Removed' }
+        if 'mapped_after_dedup' in keys:
```
Comment (ewels) [inline multiqc/modules/dedup/dedup.py:142]: You've just defined `keys` as an empty directory, so this will always be `False`..

Comment (ewels) [reply multiqc/modules/dedup/dedup.py:142]: If you get rid of the `'hide_zero_cats': False` then I don't think you need this anyway - just define all categories and the ones that aren't used will just be ignored.

**Comment by apeltzer** [comment]
Hi Phil! Thanks a bunch for fixing this :-) 

Yes, Thousands is much more appropriate for ancient DNA - we often don't even get close to millions at all, so this fits 👍

File: multiqc/modules/dedup/dedup.py (hunk 1 above)
Comment (apeltzer) [reply multiqc/modules/dedup/dedup.py]: Thanks for fixing this 🤦

**Pull request merged**